### ISSUE TITLE
Fetchmail fails on folders with German umlauts

### ISSUE THREAD

**Issue by NicolasGoeddel** [Mailu/Mailu#2996]
## Environment & Version

- Docker Compose version v2.20.0
- MailU version 2.0

## Description
When trying to fetch the folder `Entwürfe` from it results in an error. When you encode the name manually to `Entw&APw-rfe` according to the IMAP UTF-7 encoding scheme ([see here](https://stackoverflow.com/a/45787169/4239139)) it works.

## Replication Steps
- Add a fetchmail configuration, e.g. a web.de account
- Add the folder `Entwürfe` to be fetched
- Wait for the cronjob to execute

## Observed behaviour
- fetchmail runs into an ERROR

## Expected behaviour
- fetchmail should work just fine by encoding the folders to be fetched beforehand

## Logs
```
mailu-fetchmail-1  | 2023-10-22T13:26:12.475249522Z fetchmail: mailbox selection failed
mailu-fetchmail-1  | 2023-10-22T13:26:12.481254947Z fetchmail: client/server synchronization error while fetching from <private>@web.de@imap.web.de
mailu-fetchmail-1  | 2023-10-22T13:26:12.491866138Z Sleeping for 600 seconds
mailu-fetchmail-1  | 2023-10-22T13:26:12.491939658Z fetchmail: Query status=7 (ERROR)
```

**Comment by ghostwheel42** [comment]
Thanks for pointing this out. Can you create a PR to fix this in fetchmail.py?

**Comment by ghostwheel42** [comment]
As pointed out in #2997 the folders input also needs to be updated to support spaces in folder names.
Maybe this can be done by using the select2 library already used in other locations.

**Comment by nextgens** [comment]
This is probably fixed in #3407 

To test/confirm please try the following:
```
export MAILU_VERSION=pr-3407
docker compose pull
docker compose up -d
```
